content/en/blog/_posts/2022-12-01-scalable-job-tracking-ga/index.md

```diff
@@ -0,0 +1,140 @@
+ability to enable the `JobTrackingWithFinalizers` [feature gate](/docs/reference/command-line-tools-reference/feature-gates/)).
+
+Note that Jobs created before the upgrade will still be tracked using the
+legacy behavior. This is to avoid retroactively adding finalizers to running
+Pods, which might introduce race conditions.
+
+For the maximum performance on large Jobs, we recommend using the
+[Indexed completion mode](/docs/concepts/workloads/controllers/job/#completion-mode).
+In this mode, the control plane is able to track Job progress with less API
+calls.
+
+If you are a developer of operator(s) for batch, HPC, AI/ML or related workloads,
+we encourage you to use the Job API to delegate accurate progress tracking to
+Kubernetes. If there is something missing in the Job API that forces you to
+manage plain Pods, we would welcome your feedback and contributions in the
+[Working Group Batch](https://github.com/kubernetes/community/tree/master/wg-batch).
+
+### Deprecation notices
+
+During the development of the feature, the control plane added the annotation
+`batch.kubernetes.io/job-tracking` to Jobs created when the feature is enabled.
+This allowed a safe transition for older Jobs, but it was never meant to stay.
+
+In the 1.26 release, we deprecated the annotation `batch.kubernetes.io/job-tracking`
+and the control plane will stop adding it in Kubernetes 1.27.
+Along with that change, we will remove the legacy Job tracking implementation.
+As a result, the job controller will track all Jobs using finalizers and it will
+ignore from calculations all Pods that don't already have a finalizer.
+
+Before you upgrade your cluster to 1.27, we recommend that you verify that there
+are no running Jobs that don't have the annotation. Otherwise, you might observe
+the control plane recreating some Pods. We expect that this shouldn't affect
+any users, as the feature is enabled by default since Kubernetes 1.25.
+
+## What problem does the new implementation solve?
+
+Generally, Kubernetes workload controllers, such as ReplicaSet or StatefulSet,
+rely on the existence of Pods or other objects in the API to determine the
+status of the Workload and whether replacements are needed.
+For example, if a Pod that belonged to a ReplicaSet terminates or ceases to
+exist, the ReplicaSet controller needs to create a replacement Pod to satisfy
+the `.spec.replicas` field.
+
+Since its inception, the Job controller also relied on the existence of Pods in
+the API to track Job status. A Job has [completion](/docs/concepts/workloads/controllers/job/#completion-mode)
+and [failure handling](/docs/concepts/workloads/controllers/job/#handling-pod-and-container-failures)
+policies, requiring the end state of a finished Pod to determine whether to
+create a replacement Pod or mark the Job as completed or failed. As a result,
+the Job controller depended on Pods, even terminated ones, to remain in the API
+in order to keep track of the status.
```

Comment:
Maybe a link here to an issue describing in details that the other downside of this is that you need to keep the pods on the nodes, because the pod gc can't remove them, this is known fact for folks using k8s, but might not be obvious to others.